### ISSUE TITLE
IRC bot bytes vs Unicode issue on Python 3

### ISSUE THREAD

**Issue by vstinner** [buildbot/buildbot#3694]
On the #pyhon-dev IRC channel, I noticed a "b" prefix in the IRC bot message:

```
22:40 -py-bb:#python-dev- b'Build s390x SLES 3.x #10 is complete:
Warnings [\x038warnings test (warnings)\x0f] -
          http://buildbot.python.org/all/#builders/16/builds/10'
```

It seems like buildbot doesn't handle properly bytes vs Unicode in the IRC bot.

Try maybe to run your IRC bot unit tests with python3 -bb?

**Comment by seankelly** [comment]
I'm guessing there aren't unit tests for this.

**Comment by berkerpeksag** [comment]
I think this didn't get caught because the way the `send` method was patched:

```py
def patch_send(self):
    self.sent = []

    def send(msg):
        self.sent.append(msg)
    self.contact.send = send
```

The original implementation of the `send` method:

```py
def send(self, message):
    if self.muted:
        return

    if self.channel:
        self.bot.groupChat(self.channel, message)
    else:
        self.bot.chat(self.user, message)
```

`message` is encoded in the `chat` and `groupChat` methods.

**Comment by berkerpeksag** [comment]
This issue appears to be fixed in buildbot 1.0.0. I reformatted the output (copied from #python-dev channel) a little bit to make it easily displayable on GitHub:
```
-- | Notice(py-bb): Build x86 Ubuntu Shared 3.7 #111 is complete:
      Warnings [warnings test (warnings)] -
          http://buildbot.python.org/all/#builders/110/builds/111
```

However, I couldn't find a release note or a test case for this specific case.

**Comment by tardyp** [comment]
I dont think we merged anything for that. @vstinner has a patch: #3961 maybe you have it in your prod

**Comment by berkerpeksag** [comment]
You're probably right. I only saw https://github.com/python/buildmaster-config/commit/6c1f1187364822c5ca99c8bb77ed9d9c53f97568 and thought it was fixed by 1.0.0.

**Comment by rodrigc** [comment]
@zware fixed this: https://github.com/buildbot/buildbot/pull/4276
